lib/grp.gi

```diff
@@ -1693,7 +1693,8 @@ InstallMethod( Socle, "for elementary abelian groups",
 ##
 InstallMethod( Socle, "for nilpotent groups",
               [ IsGroup and IsNilpotentGroup ],
-              SUM_FLAGS,
+              RankFilter( IsGroup and IsFinite and IsNilpotentGroup )
+              - RankFilter( IsGroup and IsNilpotentGroup ),
   function(G)
```

Comment:
Looks reasonable.

Of course this constitutes a rather large drop in rank for that method, so there could be some regressions. Alas, I don't think that's a reason to block this patch -- we'll just have to watch out for regressions, and should there be any, fix them.